### DISCUSSION TITLE
How do I set the repository arguments when using the LaunchRunMutation?

### DISCUSSION THREAD

**Discussion by OwenKephart** [dagster-io/dagster#14752] [Q&A] [answered]
I'm trying to launch a run using the GraphQL API as described here: https://docs.dagster.io/concepts/dagit/graphql#launch-a-run

This has arguments for `repositoryName` and `repositoryLocationName`. Repositories are an older API, which have been superseded by the Definitions API. How do I know what to set these values to if I'm using Definitions?

**Comment by OwenKephart** [comment, marked as answer]
Previously, you could have multiple repositories per code location, but now you only have a single Definitions object per code location, making the distinction between repositoryName and repositoryLocationName confusing. To set these values you'll want:

`repositoryLocationName`: This should be set to the name of the desired _code location_. This can be found in the "Deployment" tab in the UI.

`repositoryName`: This should be set to the literal string `__repository__`.

**Reply by ogmgaye** [reply]
Out of curiosity where in their docs did you find this or how did you find this?

**Reply by OwenKephart** [reply]
Reading the code, essentially (I'm a maintainer of the OSS project). The graphQL API isn't technically public and so documentation support there isn't great, so this answer was an attempt to give people a breadcrumb in case they run into this, as it would be quite hard to find this out on your own!

**Reply by ogmgaye** [reply]
Ah thank you! Really saved my bacon on this one.
